http-clients/src/main/java/com/palantir/remoting/http/UserAgents.java

```diff
@@ -30,13 +28,16 @@ public static String getUserAgent(String userAgent, String version) {
 
     /**
      * Constructs a user agent from the {@link java.util.jar.Attributes.Name#IMPLEMENTATION_TITLE} and {@link
-     * java.util.jar.Attributes.Name#IMPLEMENTATION_VERSION} of the package of the provided class. The default value for
-     * both properties is "{@value DEFAULT_VALUE}".
+     * java.util.jar.Attributes.Name#IMPLEMENTATION_VERSION} of the package of the provided class.
+     *
+     * @throws IllegalArgumentException if the implementation title or version are not provided
      */
     public static String fromClass(Class<?> clazz) {
         Package classPackage = clazz.getPackage();
-        String userAgent = Optional.fromNullable(classPackage.getImplementationTitle()).or(DEFAULT_VALUE);
-        String version = Optional.fromNullable(classPackage.getImplementationVersion()).or(DEFAULT_VALUE);
+        String userAgent = classPackage.getImplementationTitle();
```

Comment:
trouble is that you throw an NPE, when instead i think you want to throw `IllegalArgumentException` and i think this depends on the argument that you pass in. I.e. the class you passed in is illegal because it doesnt have anything defined in the MANIFEST